### ISSUE TITLE
SDL1 Audio support

### ISSUE THREAD

**Issue by glebm** [diasurgical/devilutionX#263]
SDL1 doesn't have `SDL_QueueAudio`, which is why there is no audio at all when built with `USE_SDL1`.

In SDL1 done, playback is done by using the callback function. I'm not yet entirely sure how to implement this in SDL1.

**Comment by AJenbo** [comment]
SDL2_mixer works in a similar way, I did the initial video playback using it but the audio stuttered in a horrible way under my initial naive implementation. Doing a proper implementation appears to require you to have an intermediate buffer for decoded audio that the SDL callback can grab from. So instead I opted for disabling SDL2_mixer during video playback so that I would have access to the much simpler `SDL_QueueAudio` since it handles everything for you.

Not that `SDL_QueueAudio` is ONLY used during video playback, All in game audio is done by simply loading wave files (wave.cpp is used as a custom wave parser, but should be removed in favor of doing it all with SDL) and then played using `Mix_PlayChannel` (see dsound.cpp).

Removing the dsound abstraction and using SDL_mixer directly in sound.cpp is also planned, but there is still one missing feature that (there is a proposed solution, but I haven't had time to evaluate it yet https://github.com/diasurgical/devilutionX/pull/256).

I expect that the changes to sound.cpp would simplify adding SDL1 support a great deal so it would probably be best to deal with that first.
